components/Profile/Rank.tsx

```diff
@@ -0,0 +1,27 @@
+import getProfileRank from '@/lib/sonata/getProfileRank';
+import { useProfileProvider } from '@/providers/ProfileProvider';
+import { useEffect, useState } from 'react';
+
+const Rank = () => {
+  const [rank, setRank] = useState(0);
+  const { profile } = useProfileProvider();
+
+  useEffect(() => {
+    const init = async () => {
+      if (!profile?.verifications) return;
+      const leaderBoardRank: any = await getProfileRank(profile?.verifications);
+      setRank(leaderBoardRank);
+    };
+    if (!profile) return;
+
+    init();
+  }, [profile]);
```

Comment:
nit (optional)
- this code would be cleaner as a `useRank` hook instead of putting it inside a component file